app/data_providers/corpus/corpus_feature_group_client.py

```diff
@@ -46,6 +47,7 @@ def corpus_item_from_dict(cls, obj: Dict[str, str]) -> CorpusItemModel:
         return CorpusItemModel.parse_obj({k.lower(): v for k, v in obj.items()})
 
     @xray_recorder.capture_async('CorpusFeatureGroupClient._query_corpus_items')
+    @cached(ttl=600)
```

Comment:
what's the usual size of a candidate set? do you anticipate a significant memory increase because of caching?

Reply:
The max size of a Feature Group record is 350kb and at the moment we have ~20 candidate sets, so the memory usage is negligible. We can switch from memory to ElastiCache/Memcached if that changes.